src/main/java/com/passpnu/passwordmanager/util/JwtUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.passpnu.passwordmanager.util;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Component
+public class JwtUtil {
+    @Value("${JWT_SECRET}")
+    private String SECRET_KEY;
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public Date extractExpiration(String token) {
+        return extractClaim(token, Claims::getExpiration);
+    }
+
+    private static final int EXPIRATION = 1000 * 60 * 60 * 10;
```

Comment:
That's one way to do it :) What I thought initially is to inject it as a `@Value` from props (similar to lines 17-18)

Reply:
Fixed